src/server.ts

```diff
@@ -0,0 +1,100 @@
+//
+// Just the HTTP glue to the functionality, no implementations.
+// All actual implementations are in scanner.ts
+//
+
+import express, { Application, Request, Response } from 'express'
+import * as path from 'path'
+
+import { exec, ExecException } from 'child_process'
+
+export const app: Application = express()
+const port = 3004
+
+app.use(express.json())
+
+export interface BlockDeviceInfo {
+  readonly name: string
+  // eslint-disable-next-line no-null/no-null
+  readonly mountpoint: string | null
+  readonly children?: BlockDeviceInfo[]
+}
+
+// usb stick are ones that are mounted under /media automatically by Ubuntu
+const findBlockDevice = (
+  blockDevices: BlockDeviceInfo[]
+): BlockDeviceInfo | null => {
+  for (const blockDevice of blockDevices) {
+    if (
+      blockDevice.mountpoint &&
+      blockDevice.mountpoint.startsWith('/media/')
+    ) {
+      return blockDevice
+    }
+
+    if (blockDevice.children) {
+      const childResult = findBlockDevice(blockDevice.children)
+      if (childResult) {
+        return childResult
+      }
+    }
+  }
+
+  // eslint-disable-next-line no-null/no-null
+  return null
```

Comment:
Why not simply `return` without the `null` and save yourself having to disable-eslint elsewhere in the code where `undefined` should work just fine?

I'd test myself, but I assume that I need to have the production hardware, no?

Reply:
I'm on board with not returning null, I just got to a point where it was working and wanted to ship it.